kubernetes-client/src/main/java/io/fabric8/kubernetes/client/informers/SharedInformerFactory.java

```diff
@@ -52,6 +56,7 @@
 public class SharedInformerFactory extends BaseOperation {
   private static final Logger log = LoggerFactory.getLogger(SharedInformerFactory.class);
   private final Map<String, SharedIndexInformer> informers = new HashMap<>();
```

Comment:
Can the Map of informers be removed?  A Map doesn't seem like the right construct since you can overwrite existing entries with additional creates and the lookup is a linear scan.  Since the SharedInformer now exposes getApiTypeClass the getExistingSharedIndexInformer should just look at that.